### PR TITLE
Delete file

### PR DESCRIPTION
**Pull request by teaforthecat**
use sagas to update the form instead of waiting for a poll

### CODE REVIEW

File: client/src/saga/county/uploadBallotManifestSaga.ts [hunk 1]
```diff
@@ -43,9 +43,21 @@ function* uploadBallotManifestNetworkFail(): IterableIterator<any> {
...
+    notice.danger('There was an error deleting the file, please try again.');
+}
+
 function createUploadingBallotManifest(uploading: boolean) {
     function* uploadingBallotManifest(action: any): IterableIterator<any> {
         const data = { uploading };
+        // delete file uses this
+        if (action.data.fileType && action.data.fileType != 'bmi') {
+            return;
+        }
```
Comment (gws) [inline client/src/saga/county/uploadBallotManifestSaga.ts]: This code makes it feel like the deletion actions might have been better off mirroring the upload actions (e.g. `DELETE_CVR_...` and `DELETE_BALLOT_MANIFEST...`) and being pulled out of the `uploadBallotManifestSaga`.

Comment (teaforthecat) [reply client/src/saga/county/uploadBallotManifestSaga.ts]: I would say that the upload actions would have been better off mirroring the delete file action.

File: client/action.d.ts [hunk 2]
```diff
@@ -121,6 +122,11 @@ declare namespace Action {
         data: any;
     }
 
+    interface DeleteFileOK {
```
[the remark this answers is not in the document]
Comment (teaforthecat) [reply client/action.d.ts]: fixed

File: client/src/saga/county/uploadBallotManifestSaga.ts [hunk 3]
```diff
@@ -43,9 +43,21 @@ function* uploadBallotManifestNetworkFail(): IterableIterator<any> {
     notice.danger('Network error: failed to upload ballot manifest.');
 }
 
+function* deleteFileOK(): IterableIterator<any>{
+    notice.ok('Successfully deleted file');
+}
+
+function* deleteFileNOTOK(): IterableIterator<any>{
```
[the remark this answers is not in the document]
Comment (teaforthecat) [reply client/src/saga/county/uploadBallotManifestSaga.ts]: fixed

**Pull request merged**